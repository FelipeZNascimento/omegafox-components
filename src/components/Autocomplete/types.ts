export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
