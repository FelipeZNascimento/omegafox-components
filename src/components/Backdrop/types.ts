export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
