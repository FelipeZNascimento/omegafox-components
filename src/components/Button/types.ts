export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
