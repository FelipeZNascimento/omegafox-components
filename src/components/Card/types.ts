export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
