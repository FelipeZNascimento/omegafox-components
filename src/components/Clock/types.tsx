import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
