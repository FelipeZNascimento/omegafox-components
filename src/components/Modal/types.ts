export interface IModalProps {
  children: React.ReactNode;
  isOpen: boolean;
  title?: string;
  subtitle?: string;
  onClose: () => void;
}
