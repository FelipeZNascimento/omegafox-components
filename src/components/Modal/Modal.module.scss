@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.modal {
  @include bg-colored($color-grey1);
  @include padded(horizontal, l2);
  @include padded(vertical, l);
  @include shadowed($color-black, 0.2);

  border-radius: $border-radius;
  border-top-right-radius: 0;
  position: relative;

  &Content {
    @include padded(top, l);
    overflow-y: auto;

    &Big {
      height: auto;
      max-height: 700px;
      width: 700px;
    }

    &Small {
      height: auto;
      max-height: 600px;
      width: 500px;
    }

    &Mobile {
      height: auto;
      max-height: 80vh;
      width: 80vw;
    }
  }

  .closeIcon {
    @include bg-rgba-colored($color-black, 0.5);
    @include font-colored($color-grey1);
    @include padded(all, m);

    border-bottom-left-radius: $border-radius;
    cursor: pointer;
    position: absolute;
    right: 0;
    top: 0;
    transition: background-color $transition-time;

    &:hover {
      @include bg-rgba-colored($color-qatar-red, 1);
    }
  }

  h1,
  h2 {
    @include aligned(center);

    font-weight: normal;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  h1 {
    @include font-sized(l4);
    @include padded(all, none);
    @include marged(all, none);
  }

  h2 {
    @include font-sized(l2);
    @include padded(all, none);
    @include marged(all, none);
  }
}
