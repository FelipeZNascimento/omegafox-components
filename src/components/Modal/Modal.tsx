import React from 'react';
import { useEffect, useRef, useState } from 'react';

import classNames from 'classnames';
import { IModalProps } from './types';
import styles from './Modal.module.scss';

export const Modal = ({
  children,
  isOpen,
  title,
  subtitle,
  onClose
}: IModalProps) => {
  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);

  useEffect(() => {
    if (isOpen && !isModalOpen)
      setTimeout(function () {
        setIsModalOpen(true);
      }, 200);

    if (!isOpen && isModalOpen) {
      setTimeout(function () {
        setIsModalOpen(false);
      }, 200);
    }
  }, [isOpen, isModalOpen]);

  function useOutsideAlerter(ref: React.RefObject<HTMLDivElement>) {
    useEffect(() => {
      // Alert if clicked on outside of element
      function handleClickOutside(event: MouseEvent) {
        if (ref.current && !ref.current.contains(event.target as Node)) {
          onClose();
        }
      }
      // Bind the event listener
      document.addEventListener('mousedown', handleClickOutside);
      return () => {
        // Unbind the event listener on clean up
        document.removeEventListener('mousedown', handleClickOutside);
      };
    }, [ref]);
  }

  const backgroundClass = classNames(styles.darkBackground, {
    [styles.darkBackgroundOpen]: isModalOpen,
    [styles.darkBackgroundClosed]: !isModalOpen || !isOpen
  });

  const wrapperRef = useRef(null);
  useOutsideAlerter(wrapperRef);

  if (!isOpen && !isModalOpen) {
    return null;
  }

  return (
    <div className={backgroundClass}>
      {isModalOpen && isOpen && (
        <div className={styles.modal} ref={wrapperRef}>
          <div className={styles.closeIcon} onClick={onClose}>
            X
          </div>
          {title && <h1>{title}</h1>}
          {subtitle && <h2>{subtitle}</h2>}
          {children}
        </div>
      )}
    </div>
  );
};
