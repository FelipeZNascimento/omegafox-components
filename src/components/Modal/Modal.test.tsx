import React from 'react';
import { render } from '@testing-library/react';
import { Modal } from './Modal';
import { IModalProps } from './types';

describe('<Modal />', () => {
  const renderComponent = ({ children, isOpen, onClose }: IModalProps) =>
    render(
      <Modal isOpen={isOpen} onClose={onClose}>
        {children}
      </Modal>
    );

  it('should render', () => {
    const { container } = renderComponent({
      children: 'Content',
      isOpen: true,
      onClose: jest.fn
    });

    expect(container).toMatchSnapshot();
  });
});
