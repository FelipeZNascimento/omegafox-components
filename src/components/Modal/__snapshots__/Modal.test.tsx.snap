// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render 1`] = `
<div>
  <div
    class="darkBackground darkBackgroundClosed"
  />
</div>
`;
